server/src/main/java/org/opensearch/gateway/remote/ClusterMetadataManifest.java

```diff
@@ -366,11 +366,15 @@ public String toString() {
         return Strings.toString(MediaTypeRegistry.JSON, this);
     }
 
-    public static ClusterMetadataManifest fromXContent(XContentParser parser) throws IOException {
-        return PARSER.parse(parser, null);
+    public boolean onOrAfterCodecVersion(int codecVersion) {
+        return this.codecVersion >= codecVersion;
+    }
+
+    public static ClusterMetadataManifest fromXContentV0(XContentParser parser) throws IOException {
+        return PARSER_V0.parse(parser, null);
     }
 
-    public static ClusterMetadataManifest fromXContentV1(XContentParser parser) throws IOException {
+    public static ClusterMetadataManifest fromXContent(XContentParser parser) throws IOException {
         return PARSER_V1.parse(parser, null);
```

Comment:
create another variable , CURRENT and initialize it with PARSER_V1

Reply:
dont use PARSER_V1 directly anywhere

Reply:
Done.